modules/gapi/perf/common/gapi_imgproc_perf_tests_inl.hpp

```diff
@@ -18,6 +18,43 @@ namespace opencv_test
 
   using namespace perf;
 
+  void rgb2yuyv(const uchar* rgb_line, uchar* yuv422_line, int width)
+  {
+      CV_Assert(width % 2 == 0);
+
+      for (int i = 0; i < width; i += 2)
+      {
+          uchar r = rgb_line[i * 3    ];
+          uchar g = rgb_line[i * 3 + 1];
+          uchar b = rgb_line[i * 3 + 2];
+
+          yuv422_line[i * 2    ] = cv::saturate_cast<uchar>(-0.14713 * r - 0.28886 * g + 0.436 * b);  // U0
+          yuv422_line[i * 2 + 1] = cv::saturate_cast<uchar>(0.299   * r + 0.587   * g + 0.114 * b);  // Y0
+
+          r = rgb_line[i * 3 + 3];
+          g = rgb_line[i * 3 + 4];
+          b = rgb_line[i * 3 + 5];
+
+          yuv422_line[i * 2 + 2] = cv::saturate_cast<uchar>(0.615 * r - 0.51499 * g - 0.10001 * b);   // V0
+          yuv422_line[i * 2 + 3] = cv::saturate_cast<uchar>(0.299 * r + 0.587   * g + 0.114   * b);   // Y1
+      }
+  }
+
+  void convertRGB2YUV422Ref(const cv::Mat& in, cv::Mat &out)
+  {
+      if (out.size() != in.size())
```

Comment:
this `if` is not necessary. Just call `.create()`

Reply:
Moreover it is actually buggy to have it :)

Reply:
Fixed

Reply:
Fixed